hudi-client/hudi-client-common/src/main/java/org/apache/hudi/io/storage/HoodieHFileConfig.java

```diff
@@ -42,12 +42,12 @@
   private final boolean dropBehindCacheCompaction;
   private final Configuration hadoopConf;
   private final BloomFilter bloomFilter;
-  private final KeyValue.KVComparator hfileComparator;
+  private final CellComparator hfileComparator;
```

Comment:
are these backwards and forward compatible . i.e KVComparator is written into the HFile footer?

Reply:
Per discussion. there is a potential issue of 0.11 queries not being able to read 0.9 tables, because the `org.apache.hadoop.hbase.KeyValue.KVComparator` class is written into HFile 1.x (0.9 table) and since we now shade hbase in 0.11 as `org.apache.hudi.org.apache.hadoop.hbase.KeyValue.KVComparator`. 

If we ship the `KVComparator` class along with our jar, then we will be good.

Reply:
it can just extend `CellComparatorImpl`

Reply:
Actually, I took another approach here.  In HBase, there is already backward compatibility to transform the old comparator.  When we package the Hudi bundles, we need to make sure that `org.apache.hadoop.hbase.KeyValue$KeyComparator` (written to the metadata table HFiles in Hudi 0.9.0) is not shaded, not to break the backward compatibility logic in HBase.  The new rule is added for this.  More details can be found in the PR description.
```
<relocation>
  <pattern>org.apache.hadoop.hbase.</pattern>
  <shadedPattern>org.apache.hudi.org.apache.hadoop.hbase.</shadedPattern>
  <excludes>
    <exclude>org.apache.hadoop.hbase.KeyValue$KeyComparator</exclude>
  </excludes>
</relocation>
```

Reply:
In this way, we don't need to ship our own comparator in `org.apache.hadoop.hbase` and exclude that from shading.